### PR TITLE
New Language Support: Move on Aptos

### PR DESCRIPTION
**Pull request by ArArgon**
This PR adds support for the move language with Aptos flavor.

aptos-core: https://github.com/aptos-labs/aptos-core
Tree-sitter parser: https://github.com/aptos-labs/tree-sitter-move-on-aptos

The tree-sitter parser adopts Apache 2.0 license.

### Security

- [ ] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

**Comment by aryx** [comment]
We need the CLA to move forward. Note that the CLA does not concern the code in the tree-sitter-move-on-aptos module; it's just about the changes in the ocaml-tree-sitter-semgrep (and later semgrep) repository.

**Comment by aryx** [comment]
Would be good to add a few tests in lang/semgrep-grammars/src/semgrep-move-on-aptos/test/corpus/semgrep.txt
You can follow what the other did for the other languages and imitate.

**Review by aryx — APPROVED** [review]
LGTM. We just need the CLA and few corpus tests and we can merge this.

**Review by aryx — CHANGES_REQUESTED** [review]
request change just for those 2 things. Once done I can merge.

**Comment by ArArgon** [comment]
> LGTM. We just need the CLA and few corpus tests and we can merge this.

@aryx 
We are waiting for our legal team to give us a feedback. Once they approved, I will fill out the CLA. In the meantime, I will add some corpus tests.

Besides, what is the next step after merging this PR? When I ran `./release X --dry-run`, it suggested there should be a `semgrep-move-on-aptos` repo. 'How to add support for a new language' is out of date since section 'Publishing generated parsers', too.

**Comment by aryx** [comment]
yes this part it needs to be done at first by someone from semgrep
we need to create this empty repo.

**Comment by aryx** [comment]
created here: https://github.com/semgrep/semgrep-move-on-aptos

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
ok I've run ./release move-on-aptos and as you can see here: https://github.com/semgrep/semgrep-move-on-aptos
the C and OCaml code to parse Move is now accessible in this repository.
We now need to include this repo as a submodule in the semgrep repo and write the glue so it can be used in semgrep.

**Pull request merged**
